### PR TITLE
feat(Confirm): allow custom sizes

### LINKED ISSUE #2405: Confirm: allow custom sizes
Right now the Confirm Modal only defaults to `small` size of modal, It will be nice to have if it can utilize other Modal sizes

```jsx
// Confirm.js Line 97
<Modal {...rest} {...openProp} size='small' onClose={this.handleCancel}>
```

### PR DESCRIPTION
**Pull request by ashurbeyli**
fixes: #2405

### CODE REVIEW

**Comment by ashurbeyli** [comment]
There are 2 failed tests that is not related to this issue.

image

File: docs/app/Examples/addons/Confirm/Variations/ConfirmExampleSize.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React, { Component } from 'react'
+import { Button, Confirm } from 'semantic-ui-react'
+
+class ConfirmExampleSize extends Component {
```
Comment (layershifter) [inline docs/app/Examples/addons/Confirm/Variations/ConfirmExampleSize.js]: Let's use a in definition export, i.e. `export default class ConfirmExampleSize`

File: docs/app/Examples/addons/Confirm/Variations/ConfirmExampleSize.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  state = { open: false }
+
+  show = () => this.setState({ open: true })
```
Comment (layershifter) [inline docs/app/Examples/addons/Confirm/Variations/ConfirmExampleSize.js]: Let's add more sense to handler's name: `show` => `handleButtonClick`

File: docs/app/Examples/addons/Confirm/Variations/ConfirmExampleSize.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  render() {
+    return (
+      <div>
+        <Button onClick={this.show}>Show Large</Button>
+        <Confirm
+          open={this.state.open}
+          size="large"
+          header='This is a large confirm'
+          onCancel={this.handleCancel}
+          onConfirm={this.handleConfirm}
```
Comment (layershifter) [inline docs/app/Examples/addons/Confirm/Variations/ConfirmExampleSize.js]: Let's sort props in alphabetical order

File: src/addons/Confirm/Confirm.d.ts [hunk 4]
```diff
@@ -39,6 +39,9 @@ export interface ConfirmProps extends ModalProps {
 
   /** Whether or not the modal is visible. */
   open?: boolean;
+
+  /** A modal can vary in size. */
```
Comment (layershifter) [inline src/addons/Confirm/Confirm.d.ts]: Typo: _confirm_ can vary in size

File: src/addons/Confirm/Confirm.js [hunk 5]
```diff
@@ -46,6 +46,9 @@ class Confirm extends Component {
 
     /** Whether or not the modal is visible. */
     open: PropTypes.bool,
+
+    /** A modal can vary in size */
+    size: PropTypes.oneOf(['fullscreen', 'large', 'mini', 'small', 'tiny']),
```
Comment (layershifter) [inline src/addons/Confirm/Confirm.js]: ^

Comment (layershifter) [reply src/addons/Confirm/Confirm.js]: Let's also add a `small` as value to `defaultProps`

File: test/specs/addons/Confirm/Confirm-test.js [hunk 6]
```diff
@@ -40,13 +40,13 @@ describe('Confirm', () => {
     mapValueToProps: content => ({ content }),
   })
 
-  it('renders a small Modal', () => {
-    wrapperShallow(<Confirm />)
+  it('renders a large Modal', () => {
```
Comment (layershifter) [inline test/specs/addons/Confirm/Confirm-test.js]: Let's update this test:

```jsx
describe('children', () => {
  it('renders a Modal', () => {
    shallow(<Confirm />)
       .type()
       .should.equal(Modal)
  })
})

describe('size', () => {
  it('has "small" size by default', () => {
    shallow(<Confirm />)
      .should.have.prop('size', 'small')
  })

  _.forEach(['fullscreen', 'large', 'mini', 'small', 'tiny'], size => {
    it(`applies "${size}" size', () => {
      shallow(<Confirm size={size} />)
        .should.have.prop('size', size)
    })
  })
})
```

**Review by layershifter — CHANGES_REQUESTED** [review]
Thanks for PR, I've left some comments 👍

**Comment by layershifter** [comment]
@aabustamante yes, it's a known problem, rerun tests and everything should be fine. There are also lint issues:

image

Run `yarn run lint:fix` to fix them 👍

File: docs/app/Examples/addons/Confirm/Variations/ConfirmExampleSize.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  handleConfirm = () => this.setState({ open: false })
+  handleCancel = () => this.setState({ open: false })
+
+  render() {
+    return (
+      <div>
+        <Button onClick={this.show}>Show Large</Button>
+        <Confirm
+          open={this.state.open}
+          size="large"
```
Comment (aabustamante) [inline docs/app/Examples/addons/Confirm/Variations/ConfirmExampleSize.js]: let's keep consistency with single quotes.

Comment (aabustamante) [reply docs/app/Examples/addons/Confirm/Variations/ConfirmExampleSize.js]: Hi, @layershifter. I noticed that there is another file in this same path called `ConfirmExampleButtons`updated a year ago. This file I mention also mixes single quotes and double quotes, I remember that there are some lint tests for this, could this be indicating that these files are not in the coverage of the lint test? or maybe I'm wrong and there is no test for this, if that's the case I think we should create one, what do you think?.

**Review by aabustamante — COMMENTED** [review]
Hi, @layershifter. I am a little bit confused. I think maybe my name was not the one you wanted to mention, anyway I added some comments.

**Review by layershifter — APPROVED** [review]

File: src/addons/Confirm/Confirm.js [hunk 8]
```diff
@@ -46,12 +46,16 @@ class Confirm extends Component {
 
     /** Whether or not the modal is visible. */
     open: PropTypes.bool,
+
+    /** A modal can vary in size */
```
Comment (levithomason) [inline src/addons/Confirm/Confirm.js]: This should say `Confirm` and have a period:
```diff
-    /** A modal can vary in size */
+    /** A Confirm can vary in size. */
```

**Comment by levithomason** [comment]
Although we can add this here, all extra Confirm props are passed to the underlying Modal automatically.  If we hoist one, we should technically hoist them all.  There is an issue for that in #1169.

Again, we can merge this one for now once comments are addressed.

File: src/addons/Confirm/Confirm.js (hunk 8 above)
Comment (levithomason) [reply src/addons/Confirm/Confirm.js]: Note, I tried to update this but it is on @ashurbeyli's `master` branch which I cannot push to.

**Comment by levithomason** [comment]
Cool, was just about to make a release.  I'll wait for this to pass.

**Comment by levithomason** [comment]
Thank you!

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.77.2`

**Pull request merged**